src/status_im2/contexts/chat/photo_selector/view.cljs

```diff
@@ -107,25 +107,23 @@
       [quo/icon (if photos? :i/chevron-down :i/chevron-up)
        {:color (colors/theme-colors colors/neutral-100 colors/white)}]]]))
 
-
 (defn photo-selector
-  []
-  [:f>
-   (let [{:keys [insets]}   (rf/sub [:get-screen-params])
-         temporary-selected (reagent/atom [])] ; used when switching albums
-     (fn []
-       (let [selected        (reagent/atom []) ; currently selected
-             selected-images (rf/sub [:chats/sending-image]) ; already selected and dispatched
-             selected-album  (or (rf/sub [:camera-roll/selected-album]) (i18n/label :t/recent))]
-         (rn/use-effect
-          (fn []
-            (rf/dispatch [:chat.ui/camera-roll-get-photos 20 nil selected-album])
-            (if (seq selected-images)
-              (reset! selected (vec (vals selected-images)))
-              (reset! selected @temporary-selected)))
-          [selected-album])
-         [bottom-sheet-screen/view
-          (fn [{:keys [scroll-enabled on-scroll]}]
+  [{:keys [scroll-enabled on-scroll]}]
+  [safe-area/consumer
+   (fn [insets]
```

Comment:
Please revert this change. Reading insets from consumer causes problems on this screen. Also, I checked the PR build, the photo selector screen currently keeps doing unnecessary re-renders.

Reply:
this should be fixed properly, i'll file an issue

Reply:
@flexsurfer why don't we change the way we use insets. Since we use custom navigation bars, that means all screens need to use insets, and even many components. Reading insets from `safe-area/consumer` has mainly 2 problems: 

1. On Android, gives inaccurate big values for the first few renders 
2. Causes unnecessary re-renders (and even in some cases I faced crashes because of safe-area). 

Since safe area values are a constant, how about we read them on one of the initial screens, and then store them globally in an atom / app-db / localstorage?

Reply:
yes thats the plan

Reply:
https://github.com/status-im/status-mobile/issues/15535